online-shop/src/app/components/containers/product-details/product-details.component.ts

```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from '@angular/core';
+import { products } from 'src/app/mocks/products.mocks';
+import { Product } from 'src/app/modules/shared/types/products.types';
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.scss'],
+})
+export class ProductsDetailsComponent {
+  @Input() products: Product[] = products;
```

Comment:
Why do you need "products" as input? 
`@Input() product: Product = products[0];`
This abobe might be sufficient

Reply:
No longer need it.